crypto/jwx.go

```diff
@@ -84,6 +91,63 @@ func (client *Crypto) SignJWS(ctx context.Context, payload []byte, headers map[s
 	return signJWS(payload, headers, privateKey, detached)
 }
 
+// EncryptJWE encrypts a signed payload using the provided public key and key identifier.
+func (client *Crypto) EncryptJWE(ctx context.Context, payload []byte, headers map[string]interface{}, publicKey interface{}) (string, error) {
+	audit.Log(ctx, log.Logger(), audit.CryptoEncryptJWEEvent).Infof("Encrypting a JWE")
+	return EncryptJWE(payload, headers, publicKey)
+}
+
+// DecryptJWE decrypts a signed message using the associated private key from the kid header.
+func (client *Crypto) DecryptJWE(ctx context.Context, message string) (body []byte, headers map[string]interface{}, err error) {
+	msg, err := jwe.Parse([]byte(message))
+	if err != nil {
+		return nil, nil, err
+	}
+
+	protectedHeaders := msg.ProtectedHeaders()
+	kid := protectedHeaders.KeyID()
+	if len(kid) == 0 {
+		return nil, nil, errors.New("kid header not found")
+	}
+	privateKey, kid, err := client.getPrivateKey(kid)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	audit.Log(ctx, log.Logger(), audit.CryptoDecryptJWEEvent).Infof("Decrypting a JWE with kid: %s", kid)
+
+	body, err = jwe.Decrypt([]byte(message), protectedHeaders.Algorithm(), privateKey)
+	if err != nil {
+		return nil, nil, err
+	}
+	headers = make(map[string]interface{})
+	if uh := msg.UnprotectedHeaders(); uh != nil {
```

Comment:
do unprotected headers make sense? Could it be dangerous to have them returned (to the API client) in 1 header map? (then the client can't differentiate between protected and unprotected headers, which might lead to false assumptions).

If we don't want to complicate the API by returning them in a separate header map, do we need them for the foreseen use cases? If not, could be better to just not support it for now (until someone requests it)

Reply:
Agree not to support it for now, this also results in less code.